packages/core/admin/admin/src/hooks/useWarnIfUnsavedChanges.ts

```diff
@@ -0,0 +1,16 @@
+import { useEffect } from 'react';
+
+export function useWarnIfUnsavedChanges(enabled: boolean) {
+  useEffect(() => {
+    if (!enabled) return;
+
+    const handleBeforeUnload = (e: BeforeUnloadEvent) => {
+      e.preventDefault();
+      e.returnValue = '';
+      return '';
+    };
+
+    window.addEventListener('beforeunload', handleBeforeUnload);
```

Comment:
`beforeunload` this native browser event is displaying a native confirm alert and we don't have the control of it's content.